Tests/SnapshotTests.swift

```diff
@@ -0,0 +1,66 @@
+// Created by Cal Stephens on 12/8/21.
+// Copyright © 2021 Airbnb Inc. All rights reserved.
+
+@testable import Lottie
+import SnapshotTesting
+import XCTest
+
+// MARK: - SnapshotTests
+
+class SnapshotTests: XCTestCase {
+
+  /// Snapshots all of the sample animation JSON files visible to this test target
+  func testLottieSnapshots() {
+    #if !os(iOS)
+    // We only run snapshot tests on iOS, since running snapshot tests
+    // for macOS and tvOS would triple the number of snapshot images
+    // we have to check in to the repo.
+    XCTFail("Snapshot tests should only be ran on iOS")
+    #endif
+
+    for sampleAnimationURL in Bundle.module.urls(forResourcesWithExtension: "json", subdirectory: nil)! {
+      let sampleAnimationName = sampleAnimationURL.lastPathComponent.replacingOccurrences(of: ".json", with: "")
+
+      guard let animation = Animation.named(sampleAnimationName, bundle: .module) else {
+        continue
+      }
+
+      for percent in [0, 0.25, 0.5, 0.75, 1.0] {
+        let animationView = AnimationView(animation: animation)
+        animationView.frame.size = animation.snapshotSize
+        animationView.currentProgress = CGFloat(percent)
+
+        assertSnapshot(
+          matching: animationView,
+          as: .image,
+          named: "\(sampleAnimationName) (\(Int(percent * 100))%",
+          testName: "Sample")
```

Comment:
Do you think we should use `"\(SnapshotTests.self).\(#function)` as the test name? not sure what the intent is here with "Sample" but it feels like we may have collisions if we add more types of snapshots later on

Reply:
Sure, we can remove this. The default implementation is just `"\(#function)"`, so `Tests/__Snapshots__/Sample.Switch_States-25.png` would become `Tests/__Snapshots__/testLottieSnapshots.Switch_States-25.png`.